### ISSUE TITLE
Can't Implement maxDepth in API-Platform using YAML

### ISSUE THREAD

**Issue by fawazIbrahim** [api-platform/core#1717]
I have an entity with the following

```
/**
 * @ORM\Table(name="category")
 * @ORM\Entity(repositoryClass="AppBundle\Repository\CategoryRepository")
 */
class Category
{
     ...
     /**
     * @ORM\OneToMany(targetEntity="AppBundle\Entity\Category", mappedBy="parent")
     */
    private $children;
    ...
}
```
and i've defined the "children" as an API-Subresource using YAML:

```
AppBundle\Entity\Category:
  properties:
    children:
      subresource:
        resourceClass: 'App\Entity\Category'
```

How to set a maxDepth for the `$children` without using annotation like

```
/**
* @ApiSubresource(maxDepth=1)
*/
```

**Comment by remoteclient** [comment]
I have a similar problem with xml configuration. "maxDepth" seems to do nothing here.

**Comment by soyuka** [comment]
https://github.com/api-platform/api-platform/issues/592 ?

**Comment by Nightbr** [comment]
Hey, the maxDepth in ApiSubresource annotation is only for Subresource ROUTE maxDepth generation. 
i.e. If you have foo/{id}/bar and bar/{id}/other, Api platform will generate all subresource route such as foo/{id}/bar/{id}/other. You can chose the maxdepth of subresource route genration with the maxDepth parameter in ApiSubresource annotation.

If you want maxDepth for serialization, use directly @maxDepth annotation.

Note:  @ApiSubresource(maxDepth=1) is not released yet. Only on master branch.

**Comment by ctrl-f5** [comment]
confirming bug with xml config. maxdepth **for route generation** does not work. 
Works as expected with annotations.

**Comment by Nightbr** [comment]
Found it, We need to handle $maxDepth here https://github.com/api-platform/core/blob/master/src/Metadata/Property/Factory/ExtractorPropertyMetadataFactory.php#L151

I will make a PR today to fix this.

**Comment by Nightbr** [comment]
@ctrl-f5 @soyuka Fix in this PR https://github.com/api-platform/core/pull/2019

**Comment by Hadh** [comment]
Was this fixed?

**Comment by Nightbr** [comment]
yes, fix & merged see #1717 

Do you still have this issue?
